### PR TITLE
Add TLSv1.2 support for older Android versions

### PR DESCRIPTION
**Pull request by SandroMachado**
This pull request enables the TLSv1.2 support to Android devices with API level 16 or higher.
Android supports TLSv1.2 since version 4.1 (API level 16) but it is not enable by default, this pull request introduces a custom `SSLSocketFactory` to be used by the HTTP client that enables the TLSv1.2 support.

### CODE REVIEW

**Comment by ruimarinho** [comment]
:+1:

File: README.md [hunk 1]
```diff
@@ -1,13 +1,15 @@
 # Uphold SDK for Android
 
+[![Build Status](https://travis-ci.org/uphold/uphold-sdk-android.svg?branch=master)](https://travis-ci.org/uphold/uphold-sdk-android)
```
Comment (ruipenso) [inline README.md]: Move this to another PR.

File: README.md [hunk 2]
```diff
@@ -22,11 +24,11 @@ repositories {
 }
 
 dependencies {
-	compile ('com.github.uphold:uphold-sdk-android:0.0.2@aar') {
+	compile ('com.github.uphold:uphold-sdk-android:0.0.3@aar') {
 	    transitive = true
 	}
 	// Change to:
```
Comment (ruipenso) [inline README.md]: `// Change to the configuration bellow to use the sandbox environment.`

File: README.md [hunk 3]
```diff
@@ -22,11 +24,11 @@ repositories {
...
-	// compile ('com.github.uphold:uphold-sdk-android:0.0.2:sandboxRelease@aar') {
+	// compile ('com.github.uphold:uphold-sdk-android:0.0.3:sandboxRelease@aar') {
 	//     transitive = true
 	// }
 	// to use the sandbox environment.
```
Comment (ruipenso) [inline README.md]: Remove this.

File: README.md (hunk 2 above)
Comment (ruipenso) [inline README.md]: BTW, we could simplify this by adding the comment to the `compile` above and have the comment a little more descriptive.

**Comment by SandroMachado** [comment]
Updated @ruipenso.
